From 413349a7dd5e50d4b7cfb39c16005edf3458adde Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Tue, 10 Mar 2015 18:48:07 +0800
Subject: [PATCH 0167/1221] MLK-10397-2 ARM: dts: imx7d-19x19-arm2: add a new
 dts for sai to avoid pin conflict

Add a new dts for sai to avoid pin conflict.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/boot/dts/imx7d-19x19-arm2-sai.dts |   21 +++++++++++++++++++++
 arch/arm/boot/dts/imx7d-19x19-arm2.dts     |    2 +-
 2 files changed, 22 insertions(+), 1 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx7d-19x19-arm2-sai.dts

diff --git a/arch/arm/boot/dts/imx7d-19x19-arm2-sai.dts b/arch/arm/boot/dts/imx7d-19x19-arm2-sai.dts
new file mode 100644
index 0000000..692ee65
--- /dev/null
+++ b/arch/arm/boot/dts/imx7d-19x19-arm2-sai.dts
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7d-19x19-arm2.dts"
+
+&pinctrl_hog_1{
+	/* Pin conflict with SAI */
+	fsl,pins = <
+			MX7D_PAD_GPIO1_IO13__GPIO1_IO13  0x59
+			MX7D_PAD_SD3_STROBE__GPIO6_IO10  0x59
+		>;
+};
+
+&sai2 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx7d-19x19-arm2.dts b/arch/arm/boot/dts/imx7d-19x19-arm2.dts
index a41de81..adf7c0e 100644
--- a/arch/arm/boot/dts/imx7d-19x19-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-19x19-arm2.dts
@@ -511,7 +511,7 @@
 &sai2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_sai2>;
-	status = "okay";
+	status = "disabled";
 };
 
 &epxp {
-- 
1.7.5.4

